source/MRViewer/ImGuiMenu.cpp

```diff
@@ -1154,8 +1161,28 @@ void ImGuiMenu::drawObjectsList_()
 
     int curentDepth = 0;
     std::stack<std::shared_ptr<Object>> objDepthStack;
+
+    const float drawBoxYmin = ImGui::GetScrollY();
+    const float drawBoxYmax = drawBoxYmin + ImGui::GetWindowHeight();
+
+    int collapsedHeaderDepth = -1;
+    float skipedCursorPosY = ImGui::GetCursorPosY();
+    float lastSpacingY = 0;
+    const float menuScaling = menu_scaling();
+    const float dragDropTargetHeight = 4.f * menuScaling; // see makeDragDropTarget_(...)
+    const float itemSpacingY = ImGui::GetStyle().ItemSpacing.y;
```

Comment:
mb move all calculation in separate function?

Reply:
Better do refactoring this method. It will be done in future.